### ISSUE TITLE
allowCustom on combobox does not update ngModel on backspace or Delete

### ISSUE THREAD

**Issue by vidhikhanna** [telerik/kendo-angular#999]
## I'm submitting a...

* Bug report 

## Current behavior
* allowCustom on combobox does not update ngModel on backspace

**DESCRIPTION**

I am trying to use a kendo combobox to support custom entries. Following properties are being set:

[data] - my collection in component
[textField] - set to a property name in model of collection
[valueField] - set to a property name in model of collection
[(ngModel)] - set to selectedValue property in my component
[valuePrimitive] - "true"
[allowCustom] - "true"

The selectedValue in my component gets updated on typing a custom value not present in the given list. But when I hit backspace and tab out of control, the property does not get updated.
The problem is also reproducible if you tab on to combobox (which will auto select the text inside), hit delete button and tab out.

P.S, - Using the clear button on combobox updates the value to empty string.

## Expected behavior
* Clearing the field manually should work as typing is allowed in the combobox. 

## Minimal reproduction of the problem with instructions
* This is the [**plunker**](http://plnkr.co/edit/iLZpkkjZjb6VyfiezD3H?p=preview) to demonstrate the problem.

## Environment

Package versions:

```
+-- @angular/animations@4.4.4
+-- @angular/cli@1.5.0-beta.4
+-- @angular/common@4.4.4
+-- @angular/compiler@4.4.4
+-- @angular/compiler-cli@4.4.4
+-- @angular/core@4.4.4
+-- @angular/forms@4.4.4
+-- @angular/http@4.4.4
+-- @angular/language-service@4.4.4
+-- @angular/platform-browser@4.4.4
+-- @angular/platform-browser-dynamic@4.4.4
+-- @angular/router@4.4.4
+-- @compodoc/compodoc@1.0.1
+-- @progress/kendo-angular-buttons@2.0.0
+-- @progress/kendo-angular-dateinputs@1.2.1
+-- @progress/kendo-angular-dropdowns@1.2.2
+-- @progress/kendo-angular-excel-export@1.0.4
+-- @progress/kendo-angular-grid@1.5.0
+-- @progress/kendo-angular-inputs@1.3.1
+-- @progress/kendo-angular-intl@1.2.2
+-- @progress/kendo-angular-l10n@1.0.4
+-- @progress/kendo-angular-layout@1.0.7
+-- @progress/kendo-angular-resize-sensor@1.0.4
+-- @progress/kendo-angular-ripple@0.1.1
+-- @progress/kendo-angular-scrollview@1.0.5
+-- @progress/kendo-data-query@1.0.7
+-- @progress/kendo-drawing@1.4.0
+-- @progress/kendo-theme-default@2.42.2
+-- @progress/kendo-theme-material@0.2.2
+-- @types/jasmine@2.6.0
+-- @types/jasminewd2@2.0.3
+-- @types/node@8.0.33
+-- codelyzer@3.2.1
+-- core-js@2.5.1
+-- cors@2.8.4 extraneous
+-- event-stream@3.3.4 extraneous
+-- font-awesome@4.7.0
+-- husky@0.14.3
+-- jasmine-core@2.8.0
+-- jasmine-spec-reporter@4.2.1
+-- karma@1.7.1
+-- karma-chrome-launcher@2.2.0
+-- karma-cli@1.0.1
+-- karma-coverage-istanbul-reporter@1.3.0
+-- karma-jasmine@1.1.0
+-- karma-jasmine-html-reporter@0.2.2
+-- lint-staged@4.2.3
+-- moment@2.19.0
+-- ngx-toastr@6.4.0
+-- pdfjs-dist@1.9.638
+-- prettier@1.7.4
+-- protractor@5.1.2
+-- proxy-middleware@0.15.0 extraneous
+-- raven-js@3.18.1
+-- rxjs@5.4.3
+-- standard-version@4.2.0
+-- sweetalert2@6.10.2
+-- tiff.js@1.0.0 invalid
+-- transformation-matrix-js@2.7.1
+-- ts-node@3.3.0
+-- tslint@5.7.0
+-- typescript@2.5.3
`-- zone.js@0.8.18
```

Browser:

- Chrome (desktop) Version 60.0.3112.113 (Official Build) (64-bit)

System:
- TypeScript version: 2.5.3 
- Node version: 8.4.0  
- Platform:  Windows

**Comment by prabh-62** [comment]
With Angular 5, can we get something like `updateOn: 'blur'` option for Kendo combobox? 

More details on this Reactive Forms API: [Medium - Angular 5 with updateOn](https://netbasal.com/boosting-performance-with-the-new-updateon-option-in-angular-v5-18857279ace2)

**Comment by vidhikhanna** [comment]
Is there any update on this issue ?

**Comment by tapopov** [comment]
@vidhikhanna 
I tried reproducing this, but it seems to be working as expected. Perhaps I am missing something. Would you please provide step-by-step instructions for reproducing the issue?

**Comment by vidhikhanna** [comment]
Steps to reproduce:

1. Open the [plunker](http://plnkr.co/edit/iLZpkkjZjb6VyfiezD3H?p=preview).
2. Select an entry from the combo box, lets say "John".
Result: The label above the combobox showing the ngModel property will show the selected object as "John Smith"
3. Now focus the combobox to remove last 2 characters of John to make it Jo and tab out.
Result: the label will now show a string - "Jo" which means ngModel updated correctly as expected.
4. Now again focus the combobox and this time delete the entire string using backspace key.
Expected Result: the label value should change from "Jo" to an empty string
Actual Result: the label value does not change when the custom text is completely blanked out from keyboard.

**Comment by tapopov** [comment]
Thank you for clarifying @vidhikhanna. 
I was able to reproduce this behavior and can confirm that it is indeed not expected. We will  investigate what's causing this and provide a fix as soon as possible.

**Comment by tapopov** [comment]
Auto-closed due to commit message. Reopening until QA is passed.

**Comment by ipeshev** [comment]
Fixed in v1.2.3-dev.201711071404

**Comment by vidhikhanna** [comment]
Testing this in my project with dev package on dropdown

**Comment by vidhikhanna** [comment]
Thank you. The issue is fixed in dev package. Is there a date for 1.2.3 ?

**Comment by Igor-Baimastruc-Technosoft** [comment]
The issue is reproduced with the latest version 1.3.3

Please open [plunker](http://plnkr.co/edit/gnZG8ecFlipITqJfY5bp?p=preview)
1. Select an entry
1. Remove all letters except the first one
1. Click outside the combobox
1. Focus combobox and empty it with the Backspace
1. Click outside the combobox
**Expected Result:** the ComboBox value should change to an empty string
**Actual Result:** the ComboBox value does not change when the custom text is completely blanked out from keyboard

**Comment by tapopov** [comment]
Thanks for letting us know @Igor-Baimastruc-Technosoft . We will investigate as soon as possible.

**Comment by ipeshev** [comment]
shipped with v1.4.0

**Comment by ipeshev** [comment]
Shipped in v1.4.1
